src/StructuredLogViewer.Core/NodeQueryMatcher.cs

```diff
@@ -303,7 +322,9 @@ public SearchResult IsMatch(BaseNode node)
                     return null;
                 }
             }
-
+            
+            bool searchInName = AdditionalKeywords.ContainsKey("name");

```

Comment:
this lookup will be very expensive at runtime as it will run for 10,000,000 tree nodes. Can you declare the two booleans before the search starts?